### PR TITLE
Added new validation check for Service port protocol matching.

### PR DESCRIPTION
**Pull request by hhovsepy**
RFE https://github.com/kiali/kiali/issues/4782

Added new validation message for Service port and protocol consistency matching.

Screenshot from 2022-05-26 12-28-45
Screenshot from 2022-05-26 12-29-07

### CODE REVIEW

**Comment by hhovsepy** [comment]
Documentation PR https://github.com/kiali/kiali.io/pull/536

File: kubernetes/istio.go [hunk 1]
```diff
@@ -30,7 +30,9 @@ const (
 
 var (
 	portNameMatcher = regexp.MustCompile(`^[\-].*`)
-	portProtocols   = [...]string{"grpc", "http", "http2", "https", "mongo", "redis", "tcp", "tls", "udp", "mysql"}
+	portProtocols   = map[string][]int32{"grpc": {443}, "http": {80}, "http2": {443, 80}, "https": {443},
+		"mongo": {27017, 27018, 27019}, "redis": {6379}, "tcp": {}, "tls": {}, "udp": {},
```
Comment (lucasponce) [inline kubernetes/istio.go:34]: I'm reviewing this PR. @hhovsepy do you remember how Istio defines these mappings.
I remember this was something defined at the Envoy level, but those names/numbers were defined in some variables?

**Comment by lucasponce** [comment]
I'm reviewing this but I have to catch up a little bit more with the Istio standards, because it's pretty normal that services may define ports like HTTP with different numbers of "80" like

image

Then, this new validation is going to create a lot of new warnings, making this a little bit noise.

I guess that if Istio is managing a "table" of names/port, perhaps we could refer to that.

Anyway, I don't touch this area, I'm going to investigate this (to refresh my memories and I'll be back to this work).

**Comment by lucasponce** [comment]
@hhovsepy I think we should review this RFE, I don't think that Istio enforces to use of particular ports for particular services

https://istio.io/latest/docs/ops/deployment/requirements/#pod-requirements
https://istio.io/latest/docs/ops/configuration/traffic-management/protocol-selection/#explicit-protocol-selection

If I understand this feature, the "explicit protocol selection" helps Istio to determine if a port is of one type (to not treat that traffic as just TCP one), but it doesn't force me to use any specific port number (unless I missed something).

So, this new validation would introduce a lot of noise in general, I think the KIA0701 validation was ok because it can help if pod and service aren't in sync. But this KIA0603 would introduce more noise in normal scenarios (i.e. an HTTP port is not in the 80).

I'd recommend that we can validate the RFE to see if we missed some info, or what was the original request.

**Comment by hhovsepy** [comment]
The original issue was created after working on https://github.com/kiali/kiali/pull/4765 , could not find the suggestion which was made in the discussion thread,

**Comment by lucasponce** [comment]
@hhovsepy I'd propose to close this PR and re-open it when the investigation is progressed.
We could update the progress in the original issue https://github.com/kiali/kiali/issues/4782

**Pull request closed without merging**